### PR TITLE
shipping zone settings: fixed title for shipping methods…

### PR DESCRIPTION
**Pull request by billerr**
…without settings

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes shipping method title for methods without settings (omitting the text from the Edit & Delete links that now appear in the title erroneously). Mostly aesthetic fix.

### How to test the changes in this Pull Request:

1. In Woocommerce Settings > Shipping, enter the Edit page for a Shipping Zone.
2. Add a new Shipping method, that doesn't support `settings` or `instance-settings` in its class. 
3. The title for this method should appear as plain text and not as a link. The title shouldn't include any "Edit" or "Delete" texts (these appear as hover action links for methods WITH settings).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### Changelog entry

shipping zone settings: fixed title for shipping methods without settings

### CODE REVIEW

**Comment by claudiulodro** [comment]
Thanks for the contribution. With this change applied, for shipping method's that don't support `instance-settings`/`settings` I get a link on the shipping title that takes me to a settings page with only `'This shipping method does not have any settings to configure.'` on it.

Before:
screen shot 2018-06-08 at 11 46 08 am

After:
screen shot 2018-06-08 at 11 46 36 am

File: assets/js/admin/wc-shipping-zone-methods.js [hunk 1]
```diff
@@ -138,7 +138,7 @@
 							var $tr = view.$el.find( 'tr[data-id="' + rowData.instance_id + '"]');
 
 							if ( ! rowData.has_settings ) {
-								$tr.find( '.wc-shipping-zone-method-title a').replaceWith( $tr.find( '.wc-shipping-zone-method-title' ).text() );
+								$tr.find(".wc-shipping-zone-method-title").text(t.find(".wc-shipping-zone-method-title > a").text() );
```
Comment (claudiulodro) [inline assets/js/admin/wc-shipping-zone-methods.js]: The code style on this line doesn't match the WP style guides. Single quotes should be used and spaces should be between function parentheses. 

See the previous version for an example of how the code style should look: `$tr.find( '.wc-shipping-zone-method-title a' ).replaceWith( $tr.find( '.wc-shipping-zone-method-title' ).text() );`

**Comment by billerr** [comment]
Sorry for the wrong quotes! I must have missed this.
It's fixed and I also updated the pull request with a fix to keep the "Delete" link even for grayed out Shipping Methods without settings - in the original commit you lost the "Delete" link completely.

Regarding the issue you mention, with a link appearing instead of the method getting grayed out, I can't reproduce it. My test class only supports `'shipping-zones'` as you can see, and it shows up properly grayed out (and with just the "Delete" link). 

image

image

Could you please test it again with the new commit and see if the same keeps happening? If yes, any clues as to what may differ between your shipping method class and mine to cause this?

**Review by claudiulodro — APPROVED** [review]
Not sure what happened in testing before. Everything looks good and works well. 👍 Thanks!

**Pull request merged**